source/base/parameter_acceptor.cc

```diff
@@ -0,0 +1,241 @@
+            }
+          prm.parse_input_from_xml(is);
+        }
+      else
+        AssertThrow(false, ExcMessage("Invalid extension of parameter file. Please use .prm or .xml"));
+    }
+
+  if (output_filename != "")
+    {
+      std::ofstream outfile(output_filename.c_str());
+      Assert(outfile, ExcIO());
+      std::string extension = output_filename.substr(output_filename.find_last_of(".") + 1);
+
+      if ( extension == "prm")
+        {
+          outfile << "# Parameter file generated with " << std::endl
+                  << "# DEAL_II_GIT_BRANCH=   " << DEAL_II_GIT_BRANCH  << std::endl
+                  << "# DEAL_II_GIT_SHORTREV= " << DEAL_II_GIT_SHORTREV << std::endl;
+          Assert(output_style_for_prm_format == ParameterHandler::Text ||
+                 output_style_for_prm_format == ParameterHandler::ShortText,
+                 ExcMessage("Only Text or ShortText can be specified in output_style_for_prm_format."))
+          prm.print_parameters(outfile, output_style_for_prm_format);
+        }
+      else if (extension == "xml")
+        prm.print_parameters(outfile, ParameterHandler::XML);
+      else if (extension == "latex" || extension == "tex")
+        prm.print_parameters(outfile, ParameterHandler::LaTeX);
+      else
+        AssertThrow(false,ExcNotImplemented());
+    }
+
+  // Finally do the parsing.
+  parse_all_parameters();
+}
+
+void
+ParameterAcceptor::clear()
+{
+  class_list.clear();
+  prm.clear();
+}
+
+
+
+void ParameterAcceptor::declare_parameters(ParameterHandler &prm)
+{}
+
+
+
+void ParameterAcceptor::parse_parameters(ParameterHandler &prm)
+{}
+
+
+
+void
+ParameterAcceptor::log_info()
+{
+  deallog.push("ParameterAcceptor");
+  for (unsigned int i=0; i<class_list.size(); ++i)
+    {
+      deallog << "Class " << i << ":";
+      if (class_list[i])
+        deallog << class_list[i]->get_section_name() << std::endl;
+      else
+        deallog << " NULL" << std::endl;
+    }
+  deallog.pop();
+}
+
+void ParameterAcceptor::parse_all_parameters(ParameterHandler &prm)
+{
+  for (unsigned int i=0; i< class_list.size(); ++i)
+    if (class_list[i] != NULL)
+      {
+        class_list[i]->enter_my_subsection(prm);
+        class_list[i]->parse_parameters(prm);
+        class_list[i]->parse_parameters_call_back();
+        class_list[i]->leave_my_subsection(prm);
+      }
+}
+
+void ParameterAcceptor::declare_all_parameters(ParameterHandler &prm)
+{
+  for (unsigned int i=0; i< class_list.size(); ++i)
+    if (class_list[i] != NULL)
+      {
+        class_list[i]->enter_my_subsection(prm);
+        class_list[i]->declare_parameters(prm);
+        class_list[i]->declare_parameters_call_back();
+        class_list[i]->leave_my_subsection(prm);
+      }
+}
+
+
+std::vector<std::string>
+ParameterAcceptor::get_section_path() const
+{
+  Assert(acceptor_id < class_list.size(), ExcInternalError());
+  auto my_section_name = get_section_name();
+  bool is_absolute = my_section_name.front() == sep;
```

Comment:
maybe write as `is_absolute = (my_section_name.front() == sep);`. I had to do a double-take on the expression.

Make both variables `const`?